### PR TITLE
GFSv16.3

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
**Description**

This PR merges the GFSv16.3 updates into the `dev/gfs.v16` branch (which is replacing the `operations` branch). The version in production ended up at v16.3.4.

Changes for v16.3 include:

1. Component tag updates:
- FV3: `GFS.v16.3.0`
- GSI: `gfsda.v16.3.0`
- UPP: `upp_v8.2.0`
- UFS-UTILS: `ops-gfsv16.3.0`
- WAFS: `gfs_wafs.v6.3.1`
2. New release notes for v16.3.3 and v16.3.4.
3. Delete old ecflow definition files under `/ecf/defs`
4. Add new ecflow definition files under `/ecf/defs`
5. Resource adjustments:
- Decrease memory for `jenkfgdas_sfc` job from 80GB to 60GB
- Decrease memory for `jgdas_atmos_gempak` job from 20GB to 4GB
- Decrease memory for `jgdas_atmos_gempak_meta_ncdc` job from 20GB to 1GB
- Decrease memory for `jgfs_atmos_gempak` and `jgfs_atmos_gempak_meta` jobs from 200GB to 2GB
- Decrease memory for `jgfs_atmos_awips` and `jgfs_atmos_awips_g2` jobs from 10GB to 3GB
- Increase memory for `jgfs_atmos_wafs_blending_0p25` job from 5GB to 15GB
- Increase memory for `jgfs_atmos_wafs_grib2` job from 60GB to 80GB and increase task number from 1 to 18
- Increase memory for `jgfs_atmos_wafs_grib2_0p25` from 10GB to 80GB and increase task number from 1 to 11
- Decrease memory for `jgfs_atmos_wafs` jobs from 5GB to 1GB
- Decrease memory for `jgfs_wave_gempak` job from 10GB to 1GB
- Decrease memory for `jgfs_wave_postsbs` job from 40GB to 10GB
- Decrease memory for `jgfs_wave_prep` job from 200GB to 150GB
- Adjust R&Ds to use prior 2048M memory value for archive jobs, retain 50GB for archive jobs on WCOSS2 for now
6. New variable and cfp updates in some WAFS jobs:
- Add new `ICAO2023` variable to some WAFS job ecf scripts and set to `NO`
- Add cfp module load into some WAFS job ecf scripts and set `USE_CFP=YES`
7. Version file updates:
- in `/ecf/versions/gfs.ver`: change `obsproc_ver="1.1.0"`, `gfs_ver="v16.3"`, `crtm_ver="2.4.0"`, `bufr_ver="11.7.0"`
- in `/ecf/versions/obsproc.ver`: change `gfs_ver="v16.3"`, `obsproc_ver="1.1.0"`, `bufr_ver="11.7.0"`, `HOMEobsproc="/lfs/h1/ops/prod/packages/obsproc.v${obsproc_ver}"`
- in `/versions/build.ver`: change `crtm_ver=2.4.0`, `bufr_ver=11.7.0`, `upp_ver=8.2.0` and add `ncio_ver=1.0.0`
- in `/versions/hera.ver`, `/versions/orion.ver`, and `/versions/wcoss2.ver`: change `obsproc_run_ver=1.1.0`
- in `/versions/run.ver`: change `version=v16.3.4`, `gfs_ver=v16.3.4`, `obsproc_ver=v1.1`, `cray_pals_ver=1.1.3`, `bufr_dump_ver=1.1.2`, `crtm_ver=2.4.0` and remove `bufr_ver`
- some cleanup in version files
8. Updates in env files:
- Add `export USE_CFP=${USE_CFP:-"YES"}` in wafsgrib2 and wafsgrib20p25 block in `HERA.env`, `ORION.env`, and `WCOSS2.env`
9. Add `if [ ! -d ${COMOUT} ]; then mkdir -p $COMOUT; fi` in `jobs/JGLOBAL_ATMOS_EMCSFC_SFC_PREP`
10. Add `/$cyc` to TANK paths in `jobs/rocoto/vrfy.sh` and rename a few TANK path variables (rename `M_TANKverf` to `TANKverf_min`
11. Add missing env sourcing in `jobs/rocoto/wafsgrib2.sh` and `jobs/rocoto/wafsgrib20p25.sh`
12. Add ncio module load into `enkf_chgres_recenter_nc` modulefiles
13. Update Orion hpc-stack install location in Orion modulefiles
14. Change `nsig_ext` value from 56 to 45 in `parm/config/config.anal`
15. Add `gfsv16.3/` into historical `SATINFO` path in `parm/config/config.anal`
16. Add `,'rwmr_inc','snmr_inc','grle_inc'` into `INCREMENTS_TO_ZERO` and `INCVARS_ZERO_STRAT` settings in `parm/config/config.base*`
17. Pull in new transfer parm files from NCO
18. Some cleanup in `sorc/build_enkf_chgres_recenter_nc.sh`
19. Update `sorc/build_gsi.sh` to set `PRUNE_4NCO=YES` by default for GSI build/install
20. Change `module_fv3gfs_ncio` to `module_ncio` within `enkf_chgres_recenter_nc` code
21. Updates to `sorc/link_fv3gfs.sh`
- `FIX_DIR` updates to `fix_nco_gfsv16.3.0` fix file set locations
- add `L` option to NCO mode `LINK` command
- updates to GSI monitoring link paths (remove versions from paths)
- rename a few executables
- clean out some whitespace
22. Retire `NODESIZE` from rocoto setup scripts, no longer needed on WCOSS2 with new rocoto version that deprecated its use

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

**How Has This Been Tested?**

- [x] Clone and Build tests on Hera, Orion, WCOSS2
- [x] Cycled test on Hera, Orion, WCOSS2 (included fit2obs and MET+)
  
Refs #744 #1144

### CODE REVIEW

File: ecf/versions/gfs.ver [hunk 1]
```diff
@@ -1,17 +1,10 @@
-#### Test only remove before code delivery
-export COMPATH=/lfs/h1/ops/canned/com/ukmet:/lfs/h1/ops/canned/com/gfs:/lfs/h1/ops/canned/com/ecmwf:/lfs/h1/ops/canned/com/nam:/lfs/h1/ops/canned/com/hourly
-####
-
+export obsproc_ver="1.1.0"
```
Comment (WalterKolczynski-NOAA) [inline ecf/versions/gfs.ver:1]: Why does ecf have its own version files?

File: parm/config/config.anal [hunk 2]
```diff
@@ -122,7 +122,7 @@ if [[ $RUN_ENVIR == "emc" ]]; then
 
 #   Assimilate DO-3 Spire
     if [[ "$CDATE" -ge "2021091612" && "$CDATE" -lt "2022031612" ]]; then
-        export CONVINFO=$FIXgsi/gfsv16_historical/global_convinfo.txt.2021091612
+        export CONVINFO=$FIXgsi/gfsv16_historical/gfsv16.3/global_convinfo.txt.2021091612
```
Comment (WalterKolczynski-NOAA) [inline parm/config/config.anal:125]: If the ecf versions lists was part of the regular versions list, this might be able to use that variable.

**Review by WalterKolczynski-NOAA — APPROVED** [review]
Comments are just advisory for the future, since this is a reflection of what is in ops.

File: ecf/versions/gfs.ver (hunk 1 above)
Comment (KateFriedman-NOAA) [reply ecf/versions/gfs.ver:1]: IIRC NCO said they don't use these version files so we can remove them. They are updated here for consistency before we make the move to remove them.

File: parm/config/config.anal (hunk 2 above)
Comment (KateFriedman-NOAA) [reply parm/config/config.anal:125]: I can check with the DA team but given this is in an if-block based on date I think this is a hardcoded version and we wouldn't want to use the `gfs_ver` that changes as we move forward.

Comment (WalterKolczynski-NOAA) [reply parm/config/config.anal:125]: No, I think you're right. Didn't consider the context.

**Pull request merged**